### ISSUE TITLE
Allow hint option in aggregation

### ISSUE THREAD

**Issue by mahnunchik** [Automattic/mongoose#6251]
**What is the current behavior?**

```js
// .hint({ a: 1, b: 1, c: 1 })
X.model.aggregate(...).hint is not a function

// {hint: { a: 1, b: 1, c: 1 }}
Error: Arguments must be aggregate pipeline operators
```

**What is the expected behavior?**

https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/

hint option exists for aggregation

**Please mention your node.js, mongoose and MongoDB version.**

```bash
$ node -v
v8.9.4
$ mongo --version
MongoDB shell version v3.6.3
```

mongoose version: 4.13.10

**Comment by mahnunchik** [comment]
In mongoose 5.0.10 the same error.

**Comment by lineus** [comment]
@mahnunchik I agree that a .hint method would be nice on Model.aggregate. In the meantime, you can add the hint to the aggregation by using the .option() method. 

I wrote a mongoose example based on the example you linked to, specifically [this one](https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/#hint-an-index).

```
#!/usr/bin/env node
'use strict'

const mongoose = require('mongoose')
mongoose.connect('mongodb://localhost/test')
const Schema = mongoose.Schema

const schema = new Schema({
  category: String,
  type: String,
  qty: Number
})

schema.index({ qty: 1, type: 1 })
schema.index({ qty: 1, category: 1 })

const Food = mongoose.model('food', schema, 'food')

const foods = [
  { category: 'cake', type: 'chocolate', qty: 10 },
  { category: 'cake', type: 'ice cream', qty: 25 },
  { category: 'pie', type: 'boston cream', qty: 20 },
  { category: 'pie', type: 'blueberry', qty: 15 }
]

const pipeline = [
  { $sort: { qty: 1 } },
  { $match: { category: 'cake', qty: 10 } },
  { $sort: { type: -1 } }
]

Food.remove({}).then(() => {
  Food.create(foods).then(() => {
    Food.aggregate(pipeline)
      .option({ hint: { qty: 1, category: 1 } })
      .then((res) => {
        console.log(res)
        return mongoose.connection.close()
      }).catch((err) => {
        console.error(err)
        return mongoose.connection.close()
      })
  })
})
```
### Output running example query in Mongo Shell
```
> db.food.aggregate([ { $sort: { qty: 1 }}, { $match: { category: "cake", qty: 10  } }, { $sort: { type: -1 } } ],{ hint: { qty: 1, category: 1 } })
{ "_id" : ObjectId("5aae61cc58edbd73378d9f0a"), "category" : "cake", "type" : "chocolate", "qty" : 10, "__v" : 0 }
> 
```
### Output running my implementation in mongoose:
```
issues: ./6251.js 
[ { _id: 5aae61cc58edbd73378d9f0a,
    category: 'cake',
    type: 'chocolate',
    qty: 10,
    __v: 0 } ]
issues:
```

**Comment by mahnunchik** [comment]
Thank you!

Using `.option({ hint: { } })` helps me.

**Comment by vkarpov15** [comment]
Fixed by #6272, fix will be in 5.0.12
